make/default_compiler_options

```diff
@@ -12,8 +12,18 @@ AR = ar
 # CPPFLAGS are used for both C and C++ compilation
 CPPFLAGS = -DNO_FPRINTF_OUTPUT -pipe
 # CXXFLAGS are just used for C++
-CXXFLAGS = -Wall -I . -isystem $(EIGEN) -isystem $(BOOST) -isystem $(CVODES)/include -std=c++1y -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DBOOST_PHOENIX_NO_VARIADIC_EXPRESSION -Wno-unused-function -Wno-uninitialized
+CXXFLAGS = -Wall -I . -isystem $(EIGEN) -isystem $(BOOST) -isystem $(CVODES)/include -isystem $(OPENCL) -std=c++1y -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DBOOST_PHOENIX_NO_VARIADIC_EXPRESSION -Wno-unused-function -Wno-uninitialized -DSTAN_GPU
```

Comment:
Questions:
- are the OpenCL headers absolutely required to compile the code now?
- doesn't setting `-DSTAN_GPU` force use of GPU for all configurations?

Reply:
> are the OpenCL headers absolutely required to compile the code now?

They are only required when when we have `-DSTAN_GPU` since that includes the gpu header files in [mat.hpp](https://github.com/stan-dev/math/pull/720/files#diff-4981dc6e2a5b5152e5472ccb086f1928R246). Is there a better way to approach this?

> doesn't setting -DSTAN_GPU force use of GPU for all configurations?

Not sure what you mean by 'all configuration', but `-DSTAN_GPU` tells the compiler to include all of the gpu files. In the gpu_setup PR we also have `-DSTAN_DEVICE_CPU` in the makefile so the OpenCL compiles and runs on a CPU instead of a GPU since travis does not have GPU access.

Reply:
Got it. This isn't the way to update the makefiles. I'll help you out.

By "all configurations," I mean:
- Windows with g++
- Mac with clang++
- Mac with g++
- Linux with g++ (multiple versions)
- Linux with clang++

By setting it the way you did, it affects all of these configurations. We want to selectively do it when `STAN_GPU` is defined. In the meantime, perhaps just remove the changes to the makefile for this pull request? We can get this one in which would just be including the opencl library into the correct place?